.github/workflows/gradle-run-unit-tests.yml

```
@@ -1,21 +1,13 @@
 name: "Run Unit Tests"
 
-on:
-  merge_group:
-  pull_request:
-    types: [ opened, synchronize ] # Don't rerun on `edited` to save time
-  workflow_call:
+on: [ workflow_call ]
 
-concurrency:
-    group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
-    cancel-in-progress: true
+permissions:
+  contents: read
 
 jobs:
-  code-analysis:
-    uses: ./.github/workflows/code-analysis.yml
-  unit-tests:
+  testCoverage:
```

```suggestion
  coverage:
```

It's inside `Run Unit Tests` anyway. But if we merge ui and unit tests in the same file, it could just be `unit-tests`